packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js

```diff
@@ -196,13 +263,55 @@ const BoldChunk = (chunks) => <Typography fontWeight="bold">{chunks}</Typography
 
 const SelectedEntriesModalContent = ({ onToggle, onConfirm, onRefresh }) => {
   const { formatMessage } = useIntl();
-  const { selectedEntries, rows, isLoading, isFetching } = useTableContext();
+  const { selectedEntries, rows, onSelectRow, isLoading, isFetching } = useTableContext();
+  const [isDialogOpen, setIsDialogOpen] = React.useState(false);
+  const [isConfirmButtonLoading, setIsConfirmButtonLoading] = React.useState(false);
+  const [rowsToDisplay, setRowsToDisplay] = React.useState([]);
 
-  const selectedEntriesWithErrorsCount = rows.filter(
+  const entriesToPublish = rows
+    .filter(({ entity, errors }) => selectedEntries.includes(entity.id) && !errors)
+    .map(({ entity }) => entity.id);
+
+  const selectedEntriesWithErrorsCount = rowsToDisplay.filter(
     ({ entity, errors }) => selectedEntries.includes(entity.id) && errors
   ).length;
   const selectedEntriesWithNoErrorsCount = selectedEntries.length - selectedEntriesWithErrorsCount;
 
+  const toggleDialog = () => setIsDialogOpen((prev) => !prev);
+
+  const handleConfirmBulkPublish = async () => {
+    try {
+      toggleDialog();
+      setIsConfirmButtonLoading(true);
+      await onConfirm(entriesToPublish);
+      setIsConfirmButtonLoading(false);
+
+      const rowsToDisplayCopy = [...rowsToDisplay];
+      const update = rowsToDisplayCopy.filter((row) => {
+        if (entriesToPublish.includes(row.entity.id)) {
+          // Deselect the entries that have been published
+          onSelectRow({ name: row.entity.id, value: false });
+        }
+
+        // Remove the entries that have been published from the table
+        return !entriesToPublish.includes(row.entity.id);
+      });
+
+      setRowsToDisplay(update);
+    } catch (error) {
+      setIsConfirmButtonLoading(false);
+      toggleDialog();
+    }
+  };
+
+  React.useEffect(() => {
+    // When the api responds with data
+    if (rows.length > 0) {
+      // Update the rows to display
+      setRowsToDisplay(rows);
+    }
+  }, [rows]);
+
```

Comment:
To be completely honest I didn't want to use this effect. Originally I did 

```jsx
const [rowsToDisplay, setRowsToDisplay] = React.useState(rows);
```

which seemed to be working fine, but the tests kept failing. If anyone has a better idea I'm all ears

Reply:
This useEffect gave us the weird behaviour of the Refresh after we published some entries https://github.com/strapi/strapi/pull/17274/files#r1262211502

Reply:
The weird behaviour is this one, the first time you refresh you had back all the rows (published and not), the second time only the entry with errors as you can see in this video

https://github.com/strapi/strapi/assets/2589748/42a5683a-387b-4fa4-83ab-2852175e46ea